src/lib.rs

```diff
@@ -310,3 +318,238 @@ pub fn platform() -> Platform {
 pub fn lang() -> impl Iterator<Item = String> {
     native::lang()
 }
+
+/// Which CPU Architecture
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Arch {
+    /// ARM64
+    Aarch64,
+    /// ARM
+    Arm,
+    /// ARM BE8
+    ArmEb,
+    /// ARMv4T
+    ArmV4T,
+    /// ARMv5TE
+    ArmV5Te,
+    /// ARMv6
+    ArmV6,
+    /// ARMv7
+    ArmV7,
+    /// asm.js
+    AsmJs,
+    /// Qualcomm Hexagon
+    Hexagon,
+    /// i386
+    I386,
+    /// i586
+    I586,
+    /// i686
+    I686,
+    /// Motorola 68000 series
+    M68k,
+    /// MIPS
+    Mips,
+    /// MIPS (LE)
+    MipsEl,
+    /// MIPS64
+    Mips64,
+    /// MIPS64 (LE)
+    Mips64El,
+    /// ?
+    MipsIsa32R6,
+    /// ?
+    MipsIsa32R6El,
+    /// PowerPC
+    PowerPc,
+    /// PowerPC64
+    PowerPc64,
+    /// PowerPC64LE
+    PowerPc64Le,
+    /// 32-bit RISC-V
+    Riscv32Gc,
+    /// 64-bit RISC-V
+    Riscv64Gc,
+    /// S390x
+    S390x,
+    /// SPARC
+    Sparc,
+    /// SPARC64
+    Sparc64,
+    /// ?
+    ThumbV7A,
+    /// Thumbv7neon
+    ThumbV7Neon,
+    /// 32-bit Web Assembly
+    Wasm32,
+    /// 64-bit Web Assembly
+    Wasm64,
+    /// X86_64/Amd64
+    X86_64,
+    /// Unknown Architecture
+    Unknown(String),
+}
+
+impl std::str::FromStr for Arch {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "aarch64" | "arm64" | "aarch64_be" | "armv8b" | "armv8l" => {
+                Ok(Arch::Aarch64)
+            }
```

Comment:
With the following change applied, it will be only used on UNIX:

> This should probably just be an if statement checking `cfg(target_pointer_width = 64)`